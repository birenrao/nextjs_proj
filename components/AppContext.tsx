import { createContext } from 'react';

export interface AppContextInterface {
  movieRootPath:string;
  movieList:any;
}

const ctxt = createContext<AppContextInterface | null>(null);

export const AppContextProvider = ctxt.Provider;

export const AppContextConsumer = ctxt.Consumer;
