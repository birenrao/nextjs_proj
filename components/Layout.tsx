import * as React from 'react';
//import Link from 'next/link';
import Head from 'next/head';
import '../styles/layout.scss';
type Props = {
  title?: string;
};
const Layout: React.FunctionComponent<Props> = ({
  children,
  title = 'This is the default title',
}) => (
    <div>
      <Head>
        <title>{title}</title>
        <meta charSet="utf-8" />
        <link
            rel="shortcut icon"
            href="/static/favicon/favicon.ico"
          />
        <link href="https://fonts.googleapis.com/css?family=Roboto&display=swap" rel="stylesheet"></link>
        <meta name="viewport" content="initial-scale=1.0, width=device-width" />
      </Head>
      <div className="content">
        <header className="app-header">
          <div className="container">
            <h1>My Movies</h1>
          <div className="sub-header">
          <p>
            Select movie for more details.
          </p>
        </div>
        </div>
        </header>
        {children}
      </div>
      <footer className="app-footer">
      Â© Biren Rao 2020 - 2021 
      </footer>
    </div>
  );

export default Layout;
