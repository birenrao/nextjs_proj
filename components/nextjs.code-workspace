{
	"folders": [
		{
			"path": "/JohnLewis_Proj/DemoApp"
		},
		{
			"path": "/JohnLewis_Proj/NextJs Proj"
		}
	],
	"settings": {}
}