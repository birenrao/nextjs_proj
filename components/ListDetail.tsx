import * as React from 'react';
import { Movies } from '../interfaces';
import { AppContextConsumer } from './AppContext';
import '../styles/listDetail.scss';
type ListDetailProps = {
  item: Movies,
};

const ListDetail: React.FunctionComponent<ListDetailProps> = ({
  item: Movies,
}) => (
  <AppContextConsumer>
    {appContext => appContext && (
  <div className="container">
    <div className="movieDetails">
    <h1>{Movies.title}</h1>
    <img src={`${appContext.movieRootPath}${Movies.poster_path}`}  alt={Movies.title} />
    <p> {Movies.overview}</p>
    </div>
  </div>)}
  </AppContextConsumer>
)

export default ListDetail;
