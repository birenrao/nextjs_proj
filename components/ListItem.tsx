import * as React from 'react';
import Link from 'next/link';
import { LazyLoadImage } from 'react-lazy-load-image-component';
import 'react-lazy-load-image-component/src/effects/blur.css';
import { Movies } from '../interfaces';
import { AppContextConsumer } from './AppContext';
type Props = {
  data: Movies;
};

const ListItem: React.FunctionComponent<Props> = ({ data }) => (
  <AppContextConsumer>
    {appContext => appContext && (
  <Link href="/movies/[id]" as={`/movies/${data.id && data.id || 0}`}>
    <a>
      <h4>{data.title}</h4>
      <div className="imageContainer">
    <LazyLoadImage
      alt={data.title}
      effect="blur"
      delayTime={300}
      src={`${appContext.movieRootPath}${data.backdrop_path}`}  />
  </div>
  <p>{data.overview}</p>
    </a>
  </Link>)}
  </AppContextConsumer>
);

export default ListItem;
