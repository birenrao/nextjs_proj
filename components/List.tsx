import * as React from 'react'
import ListItem from './ListItem'
import { Movies } from '../interfaces';
import '../styles/list.scss';

type Props = {
  items: Movies[];
};

const List: React.FunctionComponent<Props> = ({ items }) => (
  <div className="grid container">
    {items && items.map(item => (
      <div className="item" key={item.id}>
        <ListItem data={item} />
      </div >
    )) || <div className="errorMSg">Error: due to some technical issues movies details are not available  </div> }
  </div>
)

export default List;
