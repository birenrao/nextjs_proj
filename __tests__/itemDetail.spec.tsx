import * as React from 'react';
import { mount, shallow } from 'enzyme';
import { Movies } from '../interfaces';
import ListDetail from '../components/ListDetail';
import { AppContextInterface, AppContextProvider } from '../components/AppContext';
import { itemsFull, results } from './items';

let tree: any;
const result: Movies = results && results[0] || [];
const sampleAppContext: AppContextInterface = {
  movieRootPath: 'https://image.tmdb.org/t/p/w300',
  movieList: itemsFull,
};
beforeEach(() => {
  tree = mount(<AppContextProvider value={sampleAppContext}>
    <ListDetail item={result} /></AppContextProvider>);
});
describe('components', () => {
  describe('List', () => {
    it('renders List detail component correctly', () => {
      expect(tree.debug()).toMatchSnapshot();
    });
  });
});
