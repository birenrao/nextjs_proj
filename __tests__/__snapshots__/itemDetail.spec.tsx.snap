// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components List renders List detail component correctly 1`] = `
"<ListDetail item={{...}}>
  <div className=\\"container\\">
    <div className=\\"movieDetails\\">
      <h1>
        Ad Astra
      </h1>
      <img src=\\"https://image.tmdb.org/t/p/w300/xBHvZcjRiWyobQ9kxBhO6B2dtRI.jpg\\" alt=\\"Ad Astra\\" />
      <p>
         
        The near future, a time when both hope and hardships drive humanity to look to the stars and beyond. While a mysterious phenomenon menaces to destroy life on planet Earth, astronaut Roy McBride undertakes a mission across the immensity of space and its many perils to uncover the truth about a lost expedition that decades before boldly faced emptiness and silence in search of the unknown.
      </p>
    </div>
  </div>
</ListDetail>"
`;
